detectron2/modeling/box_regression.py

```diff
@@ -86,7 +86,7 @@ def apply_deltas(self, deltas, boxes):
             boxes (Tensor): boxes to transform, of shape (N, 4)
         """
         deltas = deltas.float()  # ensure fp32 for decoding precision
-        boxes = boxes.to(deltas.dtype)
+        boxes = move_tensor_device_same_as_another(boxes, deltas)
```

Comment:
Is this intended? The old code does not change device.

Reply:
The new need comes from inference on multiple GPUs, but the model exported on `cuda:0` won't work for `cuda:1`. In theory we can export one model per GPU, but this is not vert convenient. I remember previously we have issue moving `pixel_mean` and `pixel_std` as well, but now it's not an issue anymore.

Regarding the performance optimization, I don't think this PR changes the behavior of any currently runnable model (`paste_masks_in_image` is already marked as `script_if_tracing`) so performance wise there's no impact. Maybe I missed something?

Reply:
About the line of code here:

* The original intention of code here is to make sure both deltas & boxes are in float32 instead of float16. No device casting is required from the code logic, although it's implicitly inserted.
* It seems that even `deltas.float()` at Line88 will be translated to `deltas.to(float, device="cpu")` in tracing and will have hard-coded device. So I think the workaround here is not sufficient.
* This whole class is scriptable. So perhaps a proper solution is to just make sure the method is scripted in tracing. The following should work:
```
def apply_deltas(self, deltas, boxes):
  @script_if_tracing
  def _do_apply_deltas(deltas, boxes, weights, clamp):
      ...

  return _do_apply_deltas(deltas, boxes, self.weights, self.scale_clamp
```